### ISSUE TITLE
TypeVars from outer class but used in inner class shouldn't be allowed

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1346]
The following should produce errors but does not:

```python
# pyright: strict

from typing import Generic, TypeVar


T = TypeVar("T")


class Outer(Generic[T]):
    class Inner:
        x: T  # Should be an error

        def f(self, x: T) -> T:
            ...  # Should be an error

        def g(self) -> None:
            y: T  # Should be an error

```

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.

**Comment by ain-soph** [comment]
If that's the case, I'm a little confused about the `dict`-like class definition, where many methods (such as `update`) require `_KT, _VT`.  
So it is for the builtin native `dict` class pyi written by Pylance (1.1.99). You can view that in the `builtins.pyi` in Pylance plugin. 

Path:  
`ms-python.vscode-pylance-2021.1.1-pre.1\dist\typeshed-fallback\stdlib\3\builtins.pyi`

```python3
class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
    # NOTE: Keyword arguments are special. If they are used, _KT must include
    #       str, but we have no way of enforcing it here.
    @overload
    def __init__(self, **kwargs: _VT) -> None: ...
    @overload
    def __init__(self, map: SupportsKeysAndGetItem[_KT, _VT], **kwargs: _VT) -> None: ...
    @overload
    def __init__(self, iterable: Iterable[Tuple[_KT, _VT]], **kwargs: _VT) -> None: ...
    def __new__(cls: Type[_T1], *args: Any, **kwargs: Any) -> _T1: ...
    def clear(self) -> None: ...
    def copy(self) -> Dict[_KT, _VT]: ...
    def popitem(self) -> Tuple[_KT, _VT]: ...
    def setdefault(self, __key: _KT, __default: _VT = ...) -> _VT: ...
    @overload
    def update(self, __m: Mapping[_KT, _VT], **kwargs: _VT) -> None: ...
```

**Comment by erictraut** [comment]
None of the classes in builtins.pyi involve class nesting (a class declared within another class), which is what this issue is about.

**Comment by ain-soph** [comment]
Sorry for the misunderstanding.
